app/src/main/java/com/nevilleantony/prototype/downloadmanager/DownloadService.java

```diff
@@ -87,28 +99,20 @@ public IBinder onBind(Intent intent) {
         return null;
     }
 
-    private void createNotification() {
-        NotificationChannel channel = new NotificationChannel(
-                TAG,
-                "Notification Channel",
-                NotificationManager.IMPORTANCE_HIGH
-        );
-        notificationManager = getSystemService(NotificationManager.class);
-        notificationManager.createNotificationChannel(channel);
-
+    private void createNotification(String fileName, String groupId) {
         Intent intentAction = new Intent(this, NotificationReceiver.class);
         intentAction.putExtra("groupId", groupId);
-        actionIntent = PendingIntent.getBroadcast(this, 1, intentAction, PendingIntent.FLAG_UPDATE_CURRENT);
+        actionIntent = PendingIntent.getBroadcast(this, notificationIdMap.get(groupId), intentAction, PendingIntent.FLAG_UPDATE_CURRENT);
 
-        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), TAG)
+        notificationBuilder.put(groupId, new NotificationCompat.Builder(getApplicationContext(), TAG)
                 .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
                 .setSmallIcon(android.R.drawable.ic_dialog_info)
-                .setContentTitle("Download")
+                .setContentTitle("Downloading " + fileName)
                 .setProgress(0, 0, false)
                 .addAction(android.R.drawable.ic_media_pause, "Pause", actionIntent)
-                .setOnlyAlertOnce(true);
-        notification = notificationBuilder.build();
-        startForeground(NOTIFICATION_ID, notification);
+                .setOnlyAlertOnce(true));
+        notification = notificationBuilder.get(groupId).build();
```

Comment:
I think this is not needed, as we are building the notification in each callback scopes anyway